lib/floe/workflow.rb

```diff
@@ -124,6 +119,12 @@ def step_nonblock
 
       step_next
       current_state.run_nonblock!
+                   .tap do
+        # Avoiding State#running? because that is potentially expensive.
+        # State#run_nonblock! already called running? via State#ready? and
+        # called State#finished -- which is what Context#state_finished? is detecting
+        context.execution["EndTime"] = context.state["FinishedTime"] if context.state_finished? && context.next_state.nil?
```

Comment:
The .tap without actually using the return looks odd, could we just do
```ruby
result = current_state.run_nonblock!
context.execution["EndTime"] = context.state["FinishedTime"] if context.state_finished? && context.next_state.nil?
result
```

Reply:
And if we have a `start_workflow!` [[Ref]](https://github.com/ManageIQ/floe/pull/206/files#r1631302239) we could call this `end_workflow!` or `finish_workflow!`

Reply:
really like this